api_test.go

```diff
@@ -319,6 +331,11 @@ func runMultipleRegistrations(configFile string, count int) (int, chan string) {
 	return count, messages
 }
 
+func clearConfig() {
+	_ = os.Remove(deviceConfigPath)
+	_ = os.Remove(privConfigFilename(deviceConfigPath))
```

Comment:
As mentioned above, having tests write out to /etc on people's machines like this isn't great.

A good way to handle this is to use the [afero package](https://github.com/spf13/afero). It provides a filesystem abstraction which can be swapped out for a fake memory based "filesystem" during tests. This makes tests much cleaner too.

An example of a simple package that uses afero like this is here: https://github.com/jumptrading/influx-spout/tree/master/config

Some key lines in that code:
* https://github.com/jumptrading/influx-spout/blob/master/config/config.go#L305
* https://github.com/jumptrading/influx-spout/blob/master/config/config.go#L278
* https://github.com/jumptrading/influx-spout/blob/master/config/config_small_test.go#L325

Use afero will allow things like `createTestConfig` to be simpler too.